IBPSA/Electrical/BaseClasses/PV/PVElectricalSingleDiodeMPP.mo

```diff
@@ -0,0 +1,193 @@
+within IBPSA.Electrical.BaseClasses.PV;
+model PVElectricalSingleDiodeMPP
+  "Analytical 5-p model for PV I-V characteristics with temperature dependency based on 5 parameters with automatic MPP control"
+  extends
+    IBPSA.Electrical.BaseClasses.PV.BaseClasses.PartialPVElectricalSingleDiode;
+
+  // Main parameters under standard conditions
+
+  Modelica.Units.SI.ElectricCurrent IPh0
+    "Photo current under standard conditions";
+  Modelica.Units.SI.ElectricCurrent IS0
+    "Saturation current under standard conditions";
+  Modelica.Units.SI.Resistance RS0
+    "Series resistance under standard conditions";
+  Modelica.Units.SI.Resistance RSh0
+    "Shunt resistance under standard conditions";
+  Real a0(unit="V") "Modified diode ideality factor under standard conditions";
+  Real w0(final unit="1")
+    "MPP auxiliary correlation coefficient under standard conditions";
+
+// Additional parameters and constants
+
+  constant Real euler=Modelica.Math.exp(1.0)
```

Comment:
You could directly use `Modelica.Constants.e`

Reply:
done